.github/workflows/rust.yml

```diff
@@ -107,12 +100,6 @@ jobs:
           path: /github/home/.cargo
           # this key equals the ones on `linux-build-lib` for re-use
           key: cargo-cache-
-      - name: Cache Rust dependencies
```

Comment:
For what it is worth the last time this CI check passed on master was https://github.com/apache/arrow-datafusion/runs/7399254770?check_suite_focus=true where it took 24m and 26 seconds (after the linux-build-lib job finished)